packages/yourloops/pages/hcp/team-leave-dialog.tsx

```diff
@@ -0,0 +1,267 @@
+  onlyMember: boolean;
+  userIsTheOnlyAdministrator: boolean;
+  buttonsDisabled: boolean;
+  handleClose: () => void;
+  handleLeaveTeam: () => Promise<void>;
+}
+
+const leaveTeamDialogClasses = makeStyles((theme: Theme) => {
+  return {
+    buttonCancel: {
+      marginRight: theme.spacing(2),
+    },
+    buttonLeaveAndDel: {
+      backgroundColor: red[500], // eslint-disable-line no-magic-numbers
+      '&:hover': {
+        backgroundColor: red[700], // eslint-disable-line no-magic-numbers
+      },
+    },
+  };
+});
+
+/**
+ * Return true if the userId is the only administrator of this team.
+ * @param team The team to test
+ * @param userId The current user id
+ */
+function isUserIsTheOnlyAdministrator(team: Team | null, userId: string): boolean {
+  if (team === null) {
+    return false;
+  }
+
+  const members = team.members ?? [];
+  let nAdmin = 0;
+  let userIsAdmin = false;
+  for (const member of members) {
+    if (member.role === "admin") {
+      nAdmin++;
+      if (member.userId === userId) {
+        userIsAdmin = true;
+      }
+    }
+  }
+  return userIsAdmin && nAdmin === 1;
+}
+
+function LeaveTeamDialogTitle(props: LeaveTeamDialogElementsProps): JSX.Element | null {
+  const { team, teamName, onlyMember, userIsTheOnlyAdministrator } = props;
+
+  if (team === null) {
+    return null;
+  }
+
+  if (onlyMember) {
+    return (
+      <DialogTitle id="team-leave-dialog-title">
+        <strong>{t("team-leave-dialog-and-del-title")}</strong>
+        <br />
+        <span id="team-leave-dialog-title-team-name">{teamName}</span>
+      </DialogTitle>
+    );
+  }
+
+  if (userIsTheOnlyAdministrator) {
+    return (
+      <DialogTitle id="team-leave-dialog-title">
+        <strong>{t("team-leave-dialog-only-admin-title")}</strong>
+        <br />
+        <span id="team-leave-dialog-title-team-name">{teamName}</span>
+      </DialogTitle>
+    );
+  }
+
+  return (
+    <DialogTitle id="team-leave-dialog-title">
+      <strong>{t("team-leave-dialog-title")}</strong>
+      <br />
+      <span id="team-leave-dialog-title-team-name">{teamName}</span>
+    </DialogTitle>
+  );
+}
+
+function LeaveTeamDialogContent(props: LeaveTeamDialogElementsProps): JSX.Element | null {
+  const { team, teamName, onlyMember, userIsTheOnlyAdministrator } = props;
+
+  if (team === null) {
+    return null;
+  }
+
+  if (onlyMember) {
+    return (
+      <DialogContent>
+        <DialogContentText id="team-leave-dialog-question">
+          <Trans i18nKey="team-leave-dialog-and-del-question" t={t} components={{ strong: <strong /> }} parent={React.Fragment}>
+            Since you are the only &quot;member&quot; in this team, {{ teamName }} will be <strong>permanently deleted</strong> if you leave it.
+          </Trans>
+        </DialogContentText>
+        <DialogContentText id="team-leave-dialog-consequences">
+          {t("team-leave-dialog-and-del-consequences")}
+        </DialogContentText>
+      </DialogContent>
+    );
```

Comment:
Same comment as on the previous method

Reply:
A PR to fix all theses little things on the team screen will follow.